### ISSUE TITLE
Document the need to set project

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#2639]
If you try to create a case without explicitly setting --project or having it set via one of the other mechanisms, you get an error, so I think it's important to document this up-front.

**Comment by gold2718** [comment]
Documentation should be clear that the requirement for a project setting is machine dependent.

**Comment by jedwards4b** [comment]
I think that this issue is incorrect - if PROJECT is defined in config_machines.xml or if PROJECT_REQUIRED is false you should not need to do this - can you confirm?

**Comment by billsacks** [comment]
@jedwards4b you're right that if PROJECT_REQUIRED is false you don't need this. As to a default PROJECT being defined: very few machines seem to have this, so in the vast majority of cases you need to set PROJECT yourself whenever PROJECT_REQUIRED is true. (And that's the way it should be, since in general different people have access to different project codes on a given machine.)

**Comment by jgfouca** [comment]
Should be addressed in web documentation.
